src/selectors/operators/numeric/BinaryNumericOperator.js

```diff
@@ -4,21 +4,24 @@ import Sequence from '../../dataTypes/Sequence';
 import Selector from '../../Selector';
 import createAtomicValue from '../../dataTypes/createAtomicValue';
 
+import YearMonthDuration from '../../dataTypes/valueTypes/YearMonthDuration';
+import DayTimeDuration from '../../dataTypes/valueTypes/DayTimeDuration';
+
 function executeOperator (kind, a, b) {
-    switch (kind) {
-        case '+':
-            return a + b;
-        case '-':
-            return a - b;
-        case '*':
-            return a * b;
-        case 'div':
-            return a / b;
-        case 'idiv':
-            return Math.trunc(a / b);
-        case 'mod':
-            return a % b;
-    }
+	switch (kind) {
```

Comment:
First comment: Nope, XPath is a dynamically typed language;
`let $a := random-number-generator()('permute')(( xs:dateTime#1, xs:string#1))("0") return $a + 1`
:no_good_man: 

Second comment:
rather something like #6, CompareExpression: precomputing the function we will need to execute.